src/Zicht/Bundle/UrlBundle/Aliasing/Doctrine/CreateAliasSubscriber.php

```diff
@@ -20,21 +20,38 @@ class CreateAliasSubscriber extends BaseSubscriber
     private $isHandling = false;
 
     /**
+     * @var array
+     */
+    private $records = [];
+
+    /**
      * @return array
      */
     public function getSubscribedEvents()
     {
         if ($this->enabled) {
-            return array(
+            return [
                 Events::postPersist,
                 Events::postUpdate,
                 Events::postFlush,
-            );
+            ];
         } else {
-            return array();
+            return [];
         }
     }
 
+    protected function addRecord($entity, array $action = [])
+    {
+        if ($entity instanceof $this->className) {
+            if (false !== ($index = array_search($entity, array_column($this->records, 0), true))) {
```

Comment:
waarom is dit beter dan ` $this->records[spl_object_hash($e->getEntity())] = $e->getEntity();`
zo een key access lijkt me sneller dan het doorzoeken van een array? of vergis ik me?